com.unity.shadergraph/Tests/Registry/TypeResolvingTest.cs

```diff
@@ -36,47 +36,33 @@ public void ResolveFromConcretizationTest()
+            NodeHandler node = m_graph.AddNode(registryKey, nodeName, m_registry);
 
             // check that the the node was added
-            var nodeReader = m_graph.GetNodeReader(nodeName);
-            bool didRead = nodeReader.TryGetField("In.Length", out GraphType.Length len);
-            Assert.IsTrue(didRead);
-
-            // EXPECT that both In and Out are concretized into length = 4 (default)
-            Assert.AreEqual(GraphType.Length.Four, len);
-            didRead = nodeReader.TryGetField("Out.Length", out len);
-            Assert.IsTrue(didRead);
-            Assert.AreEqual(GraphType.Length.Four, len);
+            var inlen = GraphTypeHelpers.GetLength(node.GetPort("In").GetTypeField());
+            Assert.AreEqual(GraphType.Length.Four, inlen);
+            var outlen = GraphTypeHelpers.GetLength(node.GetPort("Out").GetTypeField());
+            Assert.AreEqual(GraphType.Length.Four, outlen);
 
             // make In a Vec3
-            nodeWriter.SetPortField("In", GraphType.kLength, GraphType.Length.Three);
-            nodeReader = m_graph.GetNodeReader(nodeName);
-
-            // EXPECT that In now reads as a Vec3
-            didRead = nodeReader.TryGetField("In.Length", out len);
-            Assert.IsTrue(didRead);
-            Assert.AreEqual(GraphType.Length.Three, len);
+            node.GetPort("In").GetTypeField().GetSubField<GraphType.Length>(GraphType.kLength).SetData(GraphType.Length.Three);
 
-            // EXPECT that Out has not changed
-            didRead = nodeReader.TryGetField("Out.Length", out len);
-            Assert.IsTrue(didRead);
-            Assert.AreEqual(GraphType.Length.Four, len);
+            inlen = GraphTypeHelpers.GetLength(node.GetPort("In").GetTypeField());
+            Assert.AreEqual(GraphType.Length.Three, inlen);
+            outlen = GraphTypeHelpers.GetLength(node.GetPort("Out").GetTypeField());
+            Assert.AreEqual(GraphType.Length.Four, outlen);
 
             // reconcretize the node
             bool didReconcretize = m_graph.ReconcretizeNode(nodeName, m_registry);
             Assert.IsTrue(didReconcretize);
 
             // EXPECT that In is still a Vec3
-            nodeReader = m_graph.GetNodeReader(nodeName);
-            didRead = nodeReader.TryGetField("In.Length", out len);
-            Assert.IsTrue(didRead);
-            Assert.AreEqual(GraphType.Length.Three, len);
+            inlen = GraphTypeHelpers.GetLength(node.GetPort("In").GetTypeField());
+            Assert.AreEqual(GraphType.Length.Three, inlen);
 
             // EXPECT that Out has resolved into a Vec3
-            didRead = nodeReader.TryGetField("Out.Length", out len);
-            Assert.IsTrue(didRead);
-            Assert.AreEqual(GraphType.Length.Three, len);
+            outlen = GraphTypeHelpers.GetLength(node.GetPort("Out").GetTypeField());
+            Assert.AreEqual(GraphType.Length.Three, outlen);
```

Comment:
This currently fails-- Since the ports are added after the fallback resolver, the port identity doesn't exist yet, so the although the user's value is there, GetPorts doesn't find it. If a Port is stubbed in to provide the correct backing, it would need to be fully initialized first.

Some deeper considerations on concretization and the type resolver are needed to get this passing.